utils/utils_test.go

```diff
@@ -1,51 +1,470 @@
 package utils
 
 import (
+	"math/big"
+	"net/http"
+	"net/http/httptest"
+	"os"
 	"strings"
 	"testing"
 
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/stretchr/testify/assert"
 )
 
+func TestReadFile(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name        string
+		content     []byte
+		expectError bool
+	}{
+		{
+			name:    "Read existing file",
+			content: []byte("Hello, world!"),
+		},
+		{
+			name:        "File does not exist",
+			content:     nil,
+			expectError: true,
+		},
+		{
+			name:    "Empty file",
+			content: []byte(""),
+		},
+		{
+			name:    "Large file",
+			content: make([]byte, 1024*1024),
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			var filePath string
+			if tc.content != nil {
+				// Create a temporary file
+				tmpFile, err := os.CreateTemp("", "testfile")
+				assert.NoError(t, err)
+				defer os.Remove(tmpFile.Name())
+
+				// Write content to the file
+				_, err = tmpFile.Write(tc.content)
+				assert.NoError(t, err)
+
+				// Close the file
+				err = tmpFile.Close()
+				assert.NoError(t, err)
+
+				filePath = tmpFile.Name()
+			} else {
+				// Non-existent file path
+				filePath = "nonexistentfile.txt"
+			}
+
+			// Read the file using ReadFile function
+			readContent, err := ReadFile(filePath)
+			if tc.expectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tc.content, readContent)
+			}
+		})
+	}
+}
+
+func TestReadYamlConfig(t *testing.T) {
+	t.Parallel()
+
+	type Config struct {
+		Name string `yaml:"name"`
+		Age  int    `yaml:"age"`
+	}
+
+	tests := []struct {
+		name        string
+		yamlContent string
+		expectError bool
+		expected    Config
+	}{
+		{
+			name: "Valid YAML",
+			yamlContent: `
+name: John Doe
+age: 30
+`,
+			expected: Config{Name: "John Doe", Age: 30},
+		},
+		{
+			name: "Missing fields",
+			yamlContent: `
+name: Alice
+`,
+			expected: Config{Name: "Alice", Age: 0},
+		},
+		{
+			name:        "Empty YAML",
+			yamlContent: ``,
+			expected:    Config{},
+		},
```

Comment:
Not able to test with invalid values here or for `TestReadJsonConfig` since it panics. Can rewrite to return an error instead of panic so this can be tested. Consider using recover, but `log.Fatal` creates an unrecoverable panic.